packages/fastify-vite/types/index.d.ts

```diff
@@ -17,13 +17,50 @@ declare module 'fastify' {
 }
 
 type FastifyVitePlugin = FastifyPluginAsync<NonNullable<fastifyVite.FastifyViteOptions>>;
+type RouteKey = 'server'
+  | 'req'
+  | 'reply'
+  | 'head'
+  | 'state'
+  | 'data'
+  | 'firstRender'
+  | 'layout'
+  | 'getMeta'
+  | 'getData'
+  | 'onEnter'
+  | 'streaming'
+  | 'clientOnly'
+  | 'serverOnly'
+type RouteType = Partial<Record<RouteKey, any>>
+interface RendererFunctions {
+  createHtmlFunction(source: string, scope?: FastifyInstance, config?: any): (ctx: { routes: Array<RouteType>, context: any, body: any } | Record<string, any>) => Promise<any>
```

Comment:
This type definition can be difficult to read. Abstract the union into separate types, i.e.:

```ts
type T1 = { routes: Array<RouteType>, context: any, body: any }
type T2 = Record<string, any>
type Ctx = T1 | T2

{
   createHtmlFunction(source: string, scope?: FastifyInstance, config?: any): (ctx: Ctx) => Promise<any>
}
```

also prefer using template arguments with defaults to `unknown` instead of using `any` if the type is not known